pkg/operator/controller/externaldns/pod.go

```diff
@@ -194,6 +194,10 @@ func (b *externalDNSContainerBuilder) fillProviderAgnosticFields(seq int, zone s
 		args = append(args, "--ignore-hostname-annotation")
 	}
 
+	if b.externalDNS.Spec.Provider.AWS != nil && b.externalDNS.Spec.Provider.AWS.AssumeRole != nil && b.externalDNS.Spec.Provider.AWS.AssumeRole.ARN != nil {
+		args = append(args, fmt.Sprintf("--aws-assume-role=%s", *b.externalDNS.Spec.Provider.AWS.AssumeRole.ARN))
+	}
+
```

Comment:
In this function we put the flags independent from any provider. `--aws-assume-role` is AWS specific, there is a dedicated method for it: `fillAWSFields`.

Reply:
ah yes, I didn't see that. done.